### PR TITLE
re-enable MulticastInterface test on RH6, disable on RH7 and OSX

### LINKED ISSUE #34809: Test failure: System.Net.Sockets.Tests.SocketOptionNameTest/MulticastInterface_Set_AnyInterface_Succeeds
Opened on behalf of @danmosemsft

The test `System.Net.Sockets.Tests.SocketOptionNameTest/MulticastInterface_Set_AnyInterface_Succeeds` has failed.

Failure Message:
```
Waiting for received data timed out
Expected: True
Actual:   False
```

Stack Trace:
```
   at System.Net.Sockets.Tests.SocketOptionNameTest.MulticastInterface_Set_Helper(Int32 interfaceIndex) in /__w/1/s/src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs:line 132
   at System.Net.Sockets.Tests.SocketOptionNameTest.MulticastInterface_Set_AnyInterface_Succeeds() in /__w/1/s/src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs:line 91
--- End of stack trace from previous location where exception was thrown ---
```

Build : 3.0 - 20190124.1 (Core Tests)
Failing configurations:
- RedHat.6.Amd64-x64
  - Release

### PR DESCRIPTION
**Pull request by wfurt**
fixes #34809 together with infrastructure changes. 

/azp corefx-outerloop-linux
/azp corefx-outerloop-osx

### CODE REVIEW

**Comment by wfurt** [comment]
probe if updates are deployed:

/azp run corefx-outerloop-linux
/azp run corefx-outerloop-osx

File: src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs [hunk 1]
```diff
@@ -159,9 +159,11 @@ public void MulticastInterface_Set_InvalidIndex_Throws()
         [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))] // ActiveIssue: dotnet/corefx #29929
         public async Task MulticastInterface_Set_IPv6_AnyInterface_Succeeds()
         {
-            if (PlatformDetection.IsRedHatFamily6)
+            if (PlatformDetection.IsRedHatFamily7 || PlatformDetection.IsOSX)
             {
-                return; // [ActiveIssue(34809)]
+                // RH7 seems to have issues with multicast in Azure. Same code and setup can pass when executed outside of Azure.
+                // OSX lacks multicast routes. Probably not worth of fixing test to add them. Send fails with "no route" message.
```
Comment (stephentoub) [inline src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs]: Should we conditionalize the test to only run when not in CI?  I think there's a trait for that.

Comment (wfurt) [reply src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs]: maybe we should. I was thinking about it but I did not want to create new variable. I'll lookin into it.

**Pull request merged**